src/components/DappBrowser/search/SearchBar.tsx

```diff
@@ -1,49 +1,142 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { NativeSyntheticEvent, TextInput, TextInputSubmitEditingEventData } from 'react-native';
+import Animated, {
+  dispatchCommand,
+  interpolate,
+  runOnJS,
+  useAnimatedRef,
+  useAnimatedStyle,
+  useSharedValue,
+  withSpring,
+} from 'react-native-reanimated';
+import { SPRING_CONFIGS } from '@/components/animations/animationConfigs';
 import { Box } from '@/design-system';
-import Animated, { Easing, interpolate, useAnimatedKeyboard, useAnimatedStyle, withTiming } from 'react-native-reanimated';
+import { IS_IOS } from '@/env';
+import { useKeyboardHeight, useDimensions } from '@/hooks';
+import * as i18n from '@/languages';
 import { TAB_BAR_HEIGHT } from '@/navigation/SwipeNavigator';
-import { useBrowserContext } from '../BrowserContext';
+import { RAINBOW_HOME, useBrowserContext } from '../BrowserContext';
+import { GOOGLE_SEARCH_URL, HTTP, HTTPS } from '../constants';
 import { AccountIcon } from '../search-input/AccountIcon';
-import { TabButton } from '../search-input/TabButton';
 import { SearchInput } from '../search-input/SearchInput';
-import useDimensions from '@/hooks/useDimensions';
-import { IS_IOS } from '@/env';
-
-const timingConfig = {
-  duration: 500,
-  easing: Easing.bezier(0.22, 1, 0.36, 1),
-};
+import { TabButton } from '../search-input/TabButton';
+import { isValidURL } from '../utils';
 
 export const SearchBar = () => {
-  const { tabViewProgress, toggleTabView, isSearchInputFocused, searchViewProgress } = useBrowserContext();
-
-  const keyboard = useAnimatedKeyboard();
   const { width: deviceWidth } = useDimensions();
+  const { activeTabIndex, onRefresh, searchViewProgress, tabStates, tabViewProgress, tabViewVisible, updateActiveTabState } =
+    useBrowserContext();
 
-  const barStyle = useAnimatedStyle(() => ({
-    opacity: 1 - (tabViewProgress?.value ?? 0),
-    paddingLeft: withTiming(72 - 56 * (searchViewProgress?.value ?? 0), timingConfig),
-    pointerEvents: (tabViewProgress?.value ?? 0) < 1 ? 'auto' : 'none',
-    transform: [
-      {
-        translateY: interpolate(tabViewProgress?.value ?? 0, [0, 1], [0, 68], 'clamp'),
-      },
-      {
-        scale: interpolate(tabViewProgress?.value ?? 0, [0, 1], [1, 0.9], 'clamp'),
-      },
-    ],
-  }));
+  const isFocusedValue = useSharedValue(false);
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+
+  const keyboardHeight = useKeyboardHeight({ shouldListen: isFocused });
+
+  const inputRef = useAnimatedRef<TextInput>();
+
+  const tabId = useMemo(() => tabStates?.[activeTabIndex]?.uniqueId, [activeTabIndex, tabStates]);
```

Comment:
why do these two need to be memoized?

Reply:
Good call, fixed and removed similar unnecessary memoization in `BrowserTab`